pkg/blueprint/partitioning_customizations.go

```diff
@@ -0,0 +1,545 @@
+package blueprint
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"github.com/osbuild/images/pkg/pathpolicy"
+)
+
+type DiskCustomization struct {
+	// TODO: Add partition table type: gpt or dos
+	MinSize    uint64                   `json:"minsize,omitempty" toml:"minsize,omitempty"`
```

Comment:
Great idea!

Reply:
I updated/rebased https://github.com/osbuild/images/pull/1049 with this now - I mostly like it (not all of it, json custom unmarshal error handling is really bad because context is missing) but Brian dosn't, it's trade-offs, no super strong opinion here.